CRM/Export/Form/Select.php

```diff
@@ -165,39 +165,16 @@ public function getTableName() {
    */
   public function buildQuickForm() {
     //export option
-    $exportOptions = $mergeOptions = $postalMailing = [];
-    $exportOptions[] = $this->createElement('radio',
-      NULL, NULL,
-      ts('Export PRIMARY fields'),
-      self::EXPORT_ALL,
-      ['onClick' => 'showMappingOption( );']
-    );
-    $exportOptions[] = $this->createElement('radio',
-      NULL, NULL,
-      ts('Select fields for export'),
-      self::EXPORT_SELECTED,
-      ['onClick' => 'showMappingOption( );']
-    );
-
-    $mergeOptions[] = $this->createElement('radio',
-      NULL, NULL,
-      ts('Do not merge'),
-      self::EXPORT_MERGE_DO_NOT_MERGE,
-      ['onclick' => 'showGreetingOptions( );']
-    );
-    $mergeOptions[] = $this->createElement('radio',
-      NULL, NULL,
-      ts('Merge All Contacts with the Same Address'),
-      self::EXPORT_MERGE_SAME_ADDRESS,
-      ['onclick' => 'showGreetingOptions( );']
-    );
-    $mergeOptions[] = $this->createElement('radio',
-      NULL, NULL,
-      ts('Merge Household Members into their Households'),
-      self::EXPORT_MERGE_HOUSEHOLD,
-      ['onclick' => 'showGreetingOptions( );']
-    );
-
+    $exportOptions = $exportOptionsJS = $mergeOptions = $postalMailing = [];
+    $exportOptions[self::EXPORT_ALL] = ts('Export PRIMARY fields');
+    $exportOptions[self::EXPORT_SELECTED] = ts('Select fields for export');
+    $exportOptions[self::EXPORT_MERGE_DO_NOT_MERGE] = ts('Do not merge');
+    $exportOptions[self::EXPORT_MERGE_SAME_ADDRESS] = ts('Merge All Contacts with the Same Address');
+    $exportOptions[self::EXPORT_MERGE_HOUSEHOLD] = ts('Merge Household Members into their Households');
+    foreach (array_keys($exportOptions) as $key) {
+      $exportOptionsJS[$key] = ['onClick' => 'showMappingOption( );'];
```

Comment:
Some of these should be showGreetingOptions() right? Also I get `Error: Call to a member function getName() on string in HTML_QuickForm_group->onQuickFormEvent()`

Reply:
ah good point thanks for that pick up